### ISSUE TITLE
D417 false-positive

### ISSUE THREAD

**Issue by Divaltor** [charliermarsh/ruff#3578]
Hello, there is bug with enabled D417 rule (pydocstyle). Tested on **ruff 0.0.256 and Python 3.10**

```python
class Test:
  def run(self, /, arg1: int) -> None:
    """
    Some beauty description.

    Args:
      arg1: some description of arg
    """
```
```shell
$ ruff .
> D417 Missing argument description in the docstring: `self`
```

If we don't mark arguments as positional-keyword only like this
```python
class Test:
  def run(self, arg1: int) -> None:
    """
    Some beauty description.

    Args:
      arg1: some description of arg
    """
```
Then ruff gives clear log

```shell
$ ruff .
> Found 0 errors.
```
